tests/security_test.go

```diff
@@ -382,3 +383,39 @@ func runSecretTest(secretName string, secretPath string, secretKey string, expec
 
 	return nil
 }
+
+func TestPodSecurityContext(t *testing.T) {
+
+	// run driver and executor as user "nobody"
+	const uid = "65534"
+	const gid = "65534"
+
+	spark := utils.SparkOperatorInstallation{}
+	err := spark.InstallSparkOperator()
+	defer spark.CleanUp()
+
+	assert.NilError(t, err)
+
+	jobName := "mock-task-runner"
+	job := utils.SparkJob{
+		Name:     jobName,
+		Template: "spark-mock-task-runner-job.yaml",
+		Params: map[string]interface{}{
+			"args":       []string{"1", "10"},
+			"RunAsUser":  uid,
+			"RunAsGroup": gid,
+		},
+	}
+
+	err = spark.SubmitJob(&job)
+	assert.NilError(t, err)
+
+	// verify job completed successfully
+	err = spark.WaitUntilSucceeded(job)
+	assert.NilError(t, err)
+
+	// verify uid propagated to the pod
+	logContains, err := spark.DriverLogContains(job, fmt.Sprintf("myuid=%s", uid))
```

Comment:
is there a way to verify container's user via metadata instead of logs?

Reply:
yes, sounds reasonable, thanks! Added check of driver and executor pod specs in addition to driver's logs verification